python/ray/data/preprocessors/encoder.py

```diff
@@ -536,7 +536,11 @@ def update_counter(element):
         return Counter(col.value_counts(dropna=False).to_dict())
 
     def get_pd_value_counts(df: pd.DataFrame) -> List[Dict[str, Counter]]:
-        result = [{col: get_pd_value_counts_per_column(df[col]) for col in columns}]
+        for col in columns:
+            if col in df.columns.tolist():
+                result = [{col: get_pd_value_counts_per_column(df[col])}]
+            else:
+                raise ValueError(f"'{col}' does not exist as a label name in df")
```

Comment:
Could we include the list of column names that _do_ exist in the DataFrame in the error message? E.g.
```suggestion
                raise ValueError(f"Column '{col}' does not exist in DataFrame, which has columns: {df_columns}")
```